service/controller/app/resource/chart/desired.go

```diff
@@ -148,6 +153,46 @@ func generateInstall(cr v1alpha1.App) v1alpha1.ChartSpecInstall {
 	return v1alpha1.ChartSpecInstall{}
 }
 
+func getEntryURL(entries []indexcache.Entry, app, version string) (string, error) {
+	for _, e := range entries {
+		if e.Version == version {
+			if len(e.Urls) == 0 {
+				return "", microerror.Maskf(notFoundError, "no URL in index.yaml for app %#q version %#q", app, version)
+			}
+
+			return e.Urls[0], nil
+		}
+	}
+
+	return "", microerror.Maskf(notFoundError, "no app %#q in index.yaml with given version %#q", app, version)
+}
+
+func getVersionAndTarballURL(index *indexcache.Index, app, version string) (string, string, error) {
+	if index == nil || len(index.Entries) == 0 {
+		return "", "", microerror.Maskf(notFoundError, "no entries in index %#v", index)
+	}
+
+	entries, ok := index.Entries[app]
+	if !ok {
+		return "", "", microerror.Maskf(notFoundError, "no entries for app %#q in index.yaml", app)
+	}
+
+	// We first try with the full version set in .spec.version of the app CR.
+	url, err := getEntryURL(entries, app, version)
+	if err != nil {
+		// We try again without the `v` prefix. This enables us to use the
+		// Flux Image Automation controller to automatically update apps.
+		version = strings.TrimPrefix(version, "v")
+
+		url, err = getEntryURL(entries, app, version)
+		if err != nil {
+			return "", "", microerror.Mask(err)
+		}
+	}
```

Comment:
The `v` prefix thing becomes a pain here. As the hephy chart and other community catalogs sometimes use `v` in the chart version.

This lets us support this but keep the flux image automation.